### DISCUSSION TITLE
[Feature Request][overlay] Do not push content outside of a LocalOverlay

### DISCUSSION THREAD

**Discussion by jorenbroekema** [ing-bank/lion#127] [General] [answered]
Currently there is a commented out demo in `local-overlay-placement.stories.js` that shows the issue:
![image](https://user-images.githubusercontent.com/36734656/60264321-3246a700-98e3-11e9-9add-e82bca79b5e9.png)

Popper only takes care of positioning. It is not in charge of setting width and/or height limitations on the popper element just because it's too big for the viewport. The consequential behavior is that Popper enforces the viewport margin, yet the content is the max-width (viewport-width), and therefore, a piece of the content gets pushed out of the viewport entirely.

What should happen, is that we check the viewport width, deduct the viewport margin on both sides, and this is the max-width you end up with that the popper element should be limited to. Same thing for height.

We should probably add some auto overflow (optional) behavior, so that instead of the content wrapping to the next line, scroll bars are added instead so the content is not morphed. Maybe this should be split into its own enhancement issue.

Main question here for me is whether we should be doing this on the Popper-modifier layer (you can make custom ones) or on the LocalOverlay layer. On that note, we should check the open source community, maybe someone has made a custom modifier for this already. I haven't checked Popper issues for related issues either, so definitely some research should be done.

**Comment by jorenbroekema** [comment]
https://codepen.io/Falx/pen/MMrvZO  Codepen to demonstrate the issue more clearly

**Comment by erikkroes** [comment]
Hi,

We are closing this issue because we're changing the way we handle feature requests. Your issue will get the "votes needed" label, and people can add 👍 reactions to [vote and show interest](https://github.com/ing-bank/lion/issues?q=label%3A%22votes+needed%22+sort%3Areactions-%2B1-desc). It will be reopened when this feature gets picked up. 

This we way we clearly focus on bugs in our open issues. At the same time we can see what community interest there is for new features.

Thanks!

**Comment by jorenbroekema** [comment, marked as answer]
Will reopen this if it is still a problem in Popper 2
